inception-external-search-solr/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/solr/SolrSearchProvider.java

```diff
@@ -0,0 +1,330 @@
+        try {
+
+            SolrQuery query = new SolrQuery();
+            solrClient.query(aTraits.getIndexName(), query);
+            try {
+                query = new SolrQuery();
+                query.setParam("qt", aTraits.getSearchPath());
+                solrClient.query(aTraits.getIndexName(), query);
+
+                if (aQuery.isEmpty() | aQuery.equals("*:*"))
+                    aQuery = "*:*";
+                else {
+                    aQuery = aTraits.getDefaultField() + ":" + aQuery;
+                }
+
+                query.setQuery(aQuery);
+
+                //RANDOM ORDER
+                if (aTraits.isRandomOrder()) {
+                    query.setSort("random_" + aTraits.getSeed(), SolrQuery.ORDER.asc);
+                }
+
+                query.addField("*");
+                query.addField("score");
+
+                query.setHighlight(true);
+                query.addHighlightField(aTraits.getDefaultField());
+
+                query.setRows(aTraits.getResultSize());
+                query.setParam("collection", aTraits.getIndexName());
+
+
+                //RESPONSE
+                QueryResponse response = new QueryResponse();
+                try {
+                    response = solrClient.query(query);
+                    SolrDocumentList documents = response.getResults();
+
+                    for (SolrDocument document : documents) {
+                        ExternalSearchResult result = new ExternalSearchResult(aRepository,
+                                aTraits.getIndexName(), (String) document.getFirstValue("id"));
+
+                        if (!aTraits.isRandomOrder()) {
+                            double d = (float) document.getFirstValue("score");
+                            result.setScore(d);
+                        }
+
+                        fillResultWithMetadata(result, document, aTraits);
+
+                        if (response.getHighlighting().size() != 0) {
+                            List<ExternalSearchHighlight> highlights = new ArrayList<>();
+                            Map<String, Map<String, List<String>>> idHighlight =
+                                    response.getHighlighting();
+
+                            if (idHighlight.get(document.getFirstValue("id"))
+                                    .get(aTraits.getDefaultField()) != null) {
+                                for (String chaine : idHighlight.get(document.getFirstValue("id"))
+                                        .get(aTraits.getDefaultField())) {
+                                    highlights.add(new ExternalSearchHighlight(chaine));
+                                }
+                            }
+
+                            result.setHighlights(highlights);
+                        }
+                        results.add(result);
+                    }
+                    System.out.println("Found " + documents.getNumFound() + " documents");
```

Comment:
Use SLF4J for logging - even for debug or trace logging. Please do not use System.out.